### ISSUE TITLE
Task: Make this repository citeable

### ISSUE THREAD

**Issue by matt-graham** [UCL-ARC/python-tooling#330]
### What needs to be done?

We should add a `CITATION.cff` file to make it clear how to cite this repository itself, particularly as we're starting to think about advertising more widely. Relatedly we may also want to set up the Zenodo integration for the repository if we haven't already done so we get DOIs when we make a release.
```[tasklist]
### Tasks
- [x] Enable Zenodo integration
- [x] Add `CITATION.cff` file
```

**Comment by paddyroddy** [comment]
I had wondered about this recently 🤔 prompted by the [Code for Thought podcast](https://codeforthought.buzzsprout.com/1326658/14712650-en-bytesized-rse-the-citation-file-format-jason-maassen) recently

**Comment by matt-graham** [comment]
Now that GitHub allows [linking ORCIDs to profiles](https://github.blog/changelog/2024-03-13-authenticate-orcid-id/) it feels like there should be a way to automate keeping the author list on a `CITATION.cff` up to date (assuming this is based on contributor list, potentially with some filtering) 🤔

**Comment by paddyroddy** [comment]
I'm still a bit confused what the ORCiD bit is doing in `CITATION.cff.` Like it doesn't appear in the output `Roddy, P. J. (2023). SLEPLET: Slepian Scale-Discretised Wavelets in Python. Journal of Open Source Software, 8(84), 5221. https://doi.org/10.21105/joss.05221`
![image](https://github.com/UCL-ARC/python-tooling/assets/15052188/ad5ce3d8-7031-4bf0-a382-aa1f1220f23a)

**Comment by matt-graham** [comment]
The Zenodo integration automatically picks up ORCIDs from the `CITATION.cff` file I think for example (e.g. https://zenodo.org/records/10144016). I think the purpose of the file is more general than the GitHub usage.

**Comment by paddyroddy** [comment]
Maybe you're right. I know mine has picked up the ORCiD here doi.org/10.5281/zenodo.7268074, but that could have easily been from my Zenodo account (like I wouldn't need `CITATION.cff` to achieve the same). But maybe for co-authors it is necessary...

**Comment by paddyroddy** [comment]
Related zenodo/zenodo#596

**Comment by samcunliffe** [comment]
I guess we want a DOI for the pages too?

**Comment by matt-graham** [comment]
I've created a separate issue for the remain task (#411) but I think the main points here have been resolved by #340 so closing.
